### PR TITLE
Add "independence" option for ab tests

### PR DESCRIPTION
**Pull request by davidfurey**
## Why are you doing this?

This allows the randomness of test allocation between tests to be independent of each other which in theory should all us to run multiple a/b tests on the same audience segment

### CODE REVIEW

**Comment by rupertbates** [comment]
So how would this allow us to run multiple tests on the same audience segment? Just give a different indepenence number to each test on a particular segment?

**Comment by davidfurey** [comment]
Yes, you just set a different independence number for each test.  Zero means that the user will be in the same variant as dotcom.

Assuming we ran two a/b tests t1 and t2 against the same pool of users:

If independence=0 for both then all users in t1 variant will be in t2 variant, and all users in t1 control will be in t2 control.  It will be impossible to say whether any change in a metric is due to t1 or t2.

If independence=0 for t1 and independence=1 for t2 then users will be evenly distributed between (t1 variant and t2 variant), (t1 variant and t2 control), (t1 control and t2 variant) and (t2 control and t2 control).  In theory the 'other' test is just noise, and you can compare t1 variant against t1 control and have valid results.

**Comment by rupertbates** [comment]
And are Devina and Chay behind this theory?

File: assets/helpers/abtest.js [hunk 1]
```diff
@@ -125,8 +128,18 @@ function userInTest(audiences: Audiences, mvtId: number, country: IsoCountry) {
   return (mvtId > testMin) && (mvtId < testMax);
 }
 
+function randomNumber(seed: number, rounds: number): number {
```
Comment (rupertbates) [inline assets/helpers/abtest.js]: Why rounds rather than independence for the parameter name?

Comment (davidfurey) [reply assets/helpers/abtest.js]: Probably independence makes more sense.  I was originally thinking about the problem in the context of the hashing algorithms where the term 'rounds' is used for a loosely equivalent purpose.

**Comment by davidfurey** [comment]
Devina and Chay brought it up at standup so I hope they are behind it!  I thought it was sufficiently straightforward to implement that opening a PR was the easiest way of prompting discussion.  I'm hoping the @jranks123 will get involved soon too.

**Comment by jranks123** [comment]
@davidfurey @rupertbates Yes, this is important for us as it allows us to, in some cases, run a frontend test and run a support page test at the same time. 

For example, if we didn't implement this and we had a test on frontend that showed 50% of the audience EpicA and the other 50% EpicB, then they clicked through and came to the support page where we are running DesginA and DesignB, we need to randomise it so that we don't have a situation where everyone who saw EpicA sees Design A. If this happened and, for example, DesignA was the same as Design B but Epic A was better than Epic B, Design A could unfairly look like it was better. 

With the randomness, we will know that 50% of the people who saw EpicA will see DesignA, and 50% of the people who saw EpicB will see DesignA, and likewise for DesignB, thus meaning that each design will get the same audience.

**Comment by Ap0c** [comment]
@davidfurey it looks like this increased our bundle size by ~25% due to the inclusion of `sha.js` and its dependencies. Do you think there's any way we can get similar functionality but take less of a performance hit 😬?

**Comment by davidfurey** [comment]
Sorry about that - I'll have a look for a lighter-weight implementation (or algorithm)

**Pull request merged**